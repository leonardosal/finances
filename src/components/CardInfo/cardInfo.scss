.card-info {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 16px;
}
