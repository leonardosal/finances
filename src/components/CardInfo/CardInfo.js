import React from 'react';
import propTypes from 'prop-types';

import './cardInfo.scss';

import { formatValue } from '../../utils';

import Button from '../Button';

const CardInfo = ({ balance, handleClick }) => (
  <div className="card-info">
    <div>
      <h5>Saldo</h5>
      <h2>{formatValue(balance)}</h2>
    </div>
    <Button label="Adicionar Transação" onClick={handleClick} />
  </div>
);

CardInfo.propTypes = {
  balance: propTypes.number,
  handleClick: propTypes.func.isRequired,
};

CardInfo.defaultProps = {
  balance: 0,
};

export default CardInfo;
