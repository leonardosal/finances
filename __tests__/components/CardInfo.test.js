import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import CardInfo from '../../src/components/CardInfo';

it('should render card component with correct balance value', () => {
  const props = {
    balance: 1000.0,
    handleClick: () => null,
  };
  const component = mount(
    <CardInfo balance={props.label} handleClick={props.handleClick} />
  );
  expect(toJson(component)).toMatchSnapshot();
});
