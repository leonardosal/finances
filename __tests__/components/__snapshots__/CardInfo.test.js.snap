// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render card component with correct balance value 1`] = `
<CardInfo
  balance={0}
  handleClick={[Function]}
>
  <div
    className="card-info"
  >
    <div>
      <h5>
        Saldo
      </h5>
      <h2>
        R$ 0.00
      </h2>
    </div>
    <Button
      label="Adicionar Transação"
      onClick={[Function]}
      type="button"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        Adicionar Transação
      </button>
    </Button>
  </div>
</CardInfo>
`;
